### ISSUE TITLE
Data processing model

### ISSUE THREAD

**Issue by JianhuiWei7** [visurg-ai/LEMON#2]
Hi, in the example of classifying each frame into surgical or non-surgical. 

import torch
from PIL import Image
from model_loader import build_model

# Load the model
net = build_model(mode='classify')
model_path = 'Frame classification models'

# Enable multi-GPU support
net = torch.nn.DataParallel(net)
torch.backends.cudnn.benchmark = True
state = torch.load(model_path, map_location=torch.device('cpu'))
net.load_state_dict(state['net'])
net.eval()

img_path = 'path/to/your/image.jpg'
img = Image.open(img_path)
img = img.resize((224, 224))
img_tensor = torch.tensor(np.array(img)).unsqueeze(0).to('cuda')

# Extract features from the image
outputs = net(img_tensor)

Is it needed to add normalization before input to the model?

img_tensor = img_tensor / 255.0  
mean = torch.tensor([0.485, 0.456, 0.406]).view(1, 3, 1, 1) 
std = torch.tensor([0.229, 0.224, 0.225]).view(1, 3, 1, 1) 
img_tensor = (img_tensor - mean) / std 

In my toy example, results seems to be unusual without normalization.

Thanks for your contribution to the surgical foundation model!

**Comment by chengan-che** [comment]
Hi Jianhui,

Yes, during inference, normalization is necessary. You can use the following torchvision transform:

```python
torchvision.transforms.Compose([
    torchvision.transforms.ToTensor(),
    torchvision.transforms.Normalize(
        (0.4299694, 0.29676908, 0.27707579),
        (0.24373249, 0.20208984, 0.19319402)
    ),
])
```
These normalization values were computed from our dataset and differ from the standard ImageNet values to better match the surgical data distribution.

Thanks very much!

**Comment by JianhuiWei7** [comment]
Hi, jaime


```import torch
from PIL import Image
from model_loader import build_model
import torchvision.transforms as transforms

# 定义与训练时一致的预处理转换
transform = transforms.Compose([
    transforms.ToTensor(),
    transforms.Normalize(
        (0.4299694, 0.29676908, 0.27707579),
        (0.24373249, 0.20208984, 0.19319402)
    ),
])

# Load the model
net = build_model(mode='classify')
model_path = 'fold_0_model_best.pt'

# Enable multi-GPU support
net = torch.nn.DataParallel(net)
torch.backends.cudnn.benchmark = True
state = torch.load(model_path, map_location=torch.device('cpu'))
net.load_state_dict(state['net'])
net.eval()

# 图像预处理
img_path = '6b7704f64fa485f742e429f5024d8b51.png'
img = Image.open(img_path)
img = img.resize((224, 224))  # 调整为模型输入尺寸

# 应用定义好的transforms
img_tensor = transform(img)
img_tensor = img_tensor.unsqueeze(0)  # 添加批次维度

# 如果使用GPU，将数据移至GPU
device = next(net.parameters()).device
img_tensor = img_tensor.to(device)

# 模型推理
with torch.no_grad():  # 关闭梯度计算，节省内存
    outputs = net(img_tensor)

print("模型输出:", outputs)

# 获取概率和预测类别
probabilities = torch.nn.functional.softmax(outputs, dim=1)
print("类别概率:", probabilities)
_, predicted_class = torch.max(probabilities, 1)
print("预测类别:", predicted_class.item())
```

Here is my toy example, with the transform you provided.
Still, I can not get satisfactory prediction result.
Below are two simple test images. I get predicted class 1 on both of them.
Is it possible that maybe you could provide a toy example that i can try?
Thank you for your contribution to the surgical video community

<img width="3318" height="1802" alt="Image" src="https://github.com/user-attachments/assets/f0a43a88-0487-4790-9daf-0fa93f74c783" />

<img width="3266" height="1748" alt="Image" src="https://github.com/user-attachments/assets/fcf78889-a420-4a60-af5d-1909a0c9cf87" />

**Comment by JianhuiWei7** [comment]
by the way, is class index 1 or 0 stands for non-surgical content?

**Comment by chengan-che** [comment]
> by the way, is class index 1 or 0 stands for non-surgical content?

0 means surgical.

**Comment by chengan-che** [comment]
> Hi, jaime
> 
> ```
> from PIL import Image
> from model_loader import build_model
> import torchvision.transforms as transforms
> 
> # 定义与训练时一致的预处理转换
> transform = transforms.Compose([
>     transforms.ToTensor(),
>     transforms.Normalize(
>         (0.4299694, 0.29676908, 0.27707579),
>         (0.24373249, 0.20208984, 0.19319402)
>     ),
> ])
> 
> # Load the model
> net = build_model(mode='classify')
> model_path = 'fold_0_model_best.pt'
> 
> # Enable multi-GPU support
> net = torch.nn.DataParallel(net)
> torch.backends.cudnn.benchmark = True
> state = torch.load(model_path, map_location=torch.device('cpu'))
> net.load_state_dict(state['net'])
> net.eval()
> 
> # 图像预处理
> img_path = '6b7704f64fa485f742e429f5024d8b51.png'
> img = Image.open(img_path)
> img = img.resize((224, 224))  # 调整为模型输入尺寸
> 
> # 应用定义好的transforms
> img_tensor = transform(img)
> img_tensor = img_tensor.unsqueeze(0)  # 添加批次维度
> 
> # 如果使用GPU，将数据移至GPU
> device = next(net.parameters()).device
> img_tensor = img_tensor.to(device)
> 
> # 模型推理
> with torch.no_grad():  # 关闭梯度计算，节省内存
>     outputs = net(img_tensor)
> 
> print("模型输出:", outputs)
> 
> # 获取概率和预测类别
> probabilities = torch.nn.functional.softmax(outputs, dim=1)
> print("类别概率:", probabilities)
> _, predicted_class = torch.max(probabilities, 1)
> print("预测类别:", predicted_class.item())
> ```
> 
> Here is my toy example, with the transform you provided. Still, I can not get satisfactory prediction result. Below are two simple test images. I get predicted class 1 on both of them. Is it possible that maybe you could provide a toy example that i can try? Thank you for your contribution to the surgical video community
> 
> <img alt="Image" width="2000" height="1802" src="https://private-user-images.githubusercontent.com/112548025/476622003-f0a43a88-0487-4790-9daf-0fa93f74c783.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ5MzExMzQsIm5iZiI6MTc1NDkzMDgzNCwicGF0aCI6Ii8xMTI1NDgwMjUvNDc2NjIyMDAzLWYwYTQzYTg4LTA0ODctNDc5MC05ZGFmLTBmYTkzZjc0Yzc4My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgxMVQxNjQ3MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZTI1Y2UxMTM3ZjkxYWNmYzVhMzhlNTkwN2VkMTQzNGI4YWRlODNhNWEwMjE0OWJlMjU1NGEwYzg5YmY0ZmQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.wCqITnIzHMJ7SlKr4XSZ9BxK0py3Bh0a3YVWgbSmp3o"> <img alt="Image" width="2000" height="1748" src="https://private-user-images.githubusercontent.com/112548025/476622295-fcf78889-a420-4a60-af5d-1909a0c9cf87.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ5MzExMzQsIm5iZiI6MTc1NDkzMDgzNCwicGF0aCI6Ii8xMTI1NDgwMjUvNDc2NjIyMjk1LWZjZjc4ODg5LWE0MjAtNGE2MC1hZjVkLTE5MDlhMGM5Y2Y4Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgxMVQxNjQ3MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kZDEzNTEwNjQ3Y2Y1ZDc5Mzc2ODE2MTdhYWM4MzYwMGQ4YTU5NWQ3N2JkMzZmNmVkOTM5NDI3YzlmMDlmYTEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.yhxWxh9BRrZ6BeS7Rgvrz3hLx4zkWjQWVowiaGIbJV0">

Can you please provide more information about the image, such as its origin (i.e., where it came from), type of procedure it represents, and whether it is an open surgery or a minimally invasive surgery case? 

This is because our model targets minimal invasives surgeries, i have tried using several surgical images from the public minimal invasive surgery datasets like AutoLaparo, here is an example image:

![Image](https://github.com/user-attachments/assets/bbbf4465-6dbd-4144-a862-8f0c4d42b93b)

The output from using your toy code is as follows, which means the model is pretty sure that this frame is surgical, and it is correct:
类别概率: tensor([[9.9914e-01, 8.5624e-04]], device='cuda:0')
预测类别: 0

**Comment by JianhuiWei7** [comment]
The image that I use is a youtube surgical video, from https://www.youtube.com/watch?v=wzFAPKu2A8g&t=1s。The surgical procedure is Total Thyroidectomy。it is a surgery performed around the throat. Maybe it is because the surgery is open surgery?

**Comment by chengan-che** [comment]
> The image that I use is a youtube surgical video, from [https://www.youtube.com/watch?v=wzFAPKu2A8g&t=1s。The](https://www.youtube.com/watch?v=wzFAPKu2A8g&t=1s%E3%80%82The) surgical procedure is Total Thyroidectomy。it is a surgery performed around the throat. Maybe it is because the surgery is open surgery?

Hi Jianhui,

As explained in our paper, our work focuses solely on endoscopic minimally invasive surgeries, and only these are considered “surgical” by our model.

Also, a kind reminder that the paper has been renamed LEMON, which contains 4,194 videos totaling 85M frames, with two proposed downstream tasks (multi-label procedure type classification and binary surgery type classification). Please describe it accurately if you cite it. 

Thank you for your appreciation of our work.

**Comment by JianhuiWei7** [comment]
Hi, jaime.

OK, I understand the classification model now.

As for the citation thing. We hereby apologize for our mistake. We will release a new version with correct frame number and evaluation task description.

Thank you for such a great work!
